sources/packages/backend/apps/api/src/route-controllers/restriction/_tests_/e2e/restriction.institution.controller.getStudentRestrictionDetail.e2e-spec.ts

```diff
@@ -0,0 +1,154 @@
+  saveFakeStudent,
+  saveFakeStudentRestriction,
+  createFakeApplication,
+} from "@sims/test-utils";
+import { DataSource, Repository } from "typeorm";
+import {
+  Institution,
+  InstitutionLocation,
+  Application,
+  RestrictionNotificationType,
+} from "@sims/sims-db";
+
+import {
+  authorizeUserTokenForLocation,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAuthRelatedEntities,
+  getInstitutionToken,
+  InstitutionTokenTypes,
+} from "../../../../testHelpers";
+import * as request from "supertest";
+import { getUserFullName } from "../../../../utilities";
+
+describe("RestrictionInstitutionController(e2e)-getStudentRestrictions", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let collegeF: Institution;
+  let collegeC: Institution;
+  let collegeFLocation: InstitutionLocation;
+  let collegeCLocation: InstitutionLocation;
+  let applicationRepo: Repository<Application>;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+
+    const { institution } = await getAuthRelatedEntities(
+      appDataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeF = institution;
+    collegeFLocation = createFakeInstitutionLocation(collegeF);
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+    applicationRepo = appDataSource.getRepository(Application);
+  });
+
+  it("Should return forbidden 403 error code when a non BC Public Institution User tries to get student restriction by student restriction id.", async () => {
+    // Arrange
+    const { institution } = await getAuthRelatedEntities(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    collegeC = institution;
+    collegeCLocation = createFakeInstitutionLocation(collegeC);
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCLocation,
+    );
+    const student = await saveFakeStudent(appDataSource);
+    const application = createFakeApplication({
+      location: collegeCLocation,
+      student,
+    });
+    await applicationRepo.save(application);
+    const studentRestriction = await saveFakeStudentRestriction(
+      appDataSource,
+      { student, application, creator: student.user },
+      {
+        restrictionCategory: "Verification",
+        notificationType: RestrictionNotificationType.NoEffect,
+      },
+    );
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    const endpoint = `/institutions/restriction/student/${student.id}/studentRestriction/${studentRestriction.id}`;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.FORBIDDEN);
```

Comment:
Please check the complete return message. The other simular E2E tests can be used as a reference.

Reply:
```ts
.expect(HttpStatus.FORBIDDEN)
      .expect({
        statusCode: HttpStatus.FORBIDDEN,
        message: INSTITUTION_BC_PUBLIC_ERROR_MESSAGE,
        error: "Forbidden",
      });
```